lib/puppet/type.rb

```diff
@@ -1316,6 +1318,7 @@ def pathbuilder
       and the second run will log the edit made by Puppet.)"
 
     validate do |list|
+      Puppet.deprecation_warning(_("The `audit` metaparameter is deprecated and will be ignored in a future release."))
```

Comment:
After talking with @ahpook the desired behavior is to:

1. Document that `audit` metaparam is deprecated (done)
2. By default do not emit deprecation warnings if they appear in a manifest
3. When running with `strict` mode enabled, emit deprecation warnings for each resource using `audit` including file and line numbers, so users know where to look to remove them.

/cc @nfagerlund @jtappa